meter.go

```diff
@@ -53,6 +53,16 @@ func (m *Meter) Snapshot() Snapshot {
 	return m.snapshot
 }
 
+// Clear sets accumulator, total and rate to zero and last update to current time.
+func (m *Meter) Clear() {
+	globalSweeper.snapshotMu.RLock()
+	defer globalSweeper.snapshotMu.RUnlock()
+	atomic.StoreUint64(&m.accumulator, 0)
```

Comment:
Unfortunately, this won't work:

1. Thread A calls `Mark`, transitioning the meter from idle to active. The meter is now written to the sweeper's registration channel.
2. Thread B calls `Clear`, resetting the accumulator to 0.
3. Thread A calls `Mark`, registering the meter _again_.

The meter will be registered twice.

A simple solution is to deduplicate meters inside the sweeper but I was trying to avoid keeping a map inside the sweeper to save memory. On the other hand, this map will be linear in the number of active meters, not linear in the number of total meters, so it's not _that_ bad.

Reply:
How about storing the info in the Meter itself whether it was registered or not?

Reply:
Yeah, that's probably fine. it's an extra word per meter but not the end of the world.

Reply:
I managed to write a reproducible race test for this case and pushed a fix.

Reply:
Done